retrieval/discovery/consul.go

```diff
@@ -95,6 +96,17 @@ func NewConsulDiscovery(conf *config.ConsulSDConfig) *ConsulDiscovery {
 		scrapedServices: map[string]struct{}{},
 		services:        map[string]*consulService{},
 	}
+	// If the datacenter isn't set in the clientConf, let's get it from the local Consul agent
+	// (Consul default is to use local node's datacenter if one isn't given for a query)
+	if clientConf.Datacenter == "" {
+		info, err := client.Agent().Self()
+		if err != nil {
+			panic(fmt.Errorf("discovery.NewConsulDiscovery: %s", err))
```

Comment:
I noticed that and the existing code is not leading with a good example here (even though the previous panic will technically never happen).

We already added constructor errors with Kubernetes, so I plan to build something around that. We can leave it as is for this PR.